### ISSUE TITLE
A great sandbox idea

### ISSUE THREAD

**Issue by keskad** [feulf/raintpl3#115]
Hello.

I just got a brilliant idea in my head, so I must write it here.

It's possible to parse PHP code after compilation (using PHP's tokenizer) and check for unwanted functions.

Example usage of tokenizer to get all defined classes in php file:
https://github.com/webnull/panthera/blob/master/lib/modules/autoloader.tools.module.php#L86

To get a hermetic sandbox there can be dynamic function names disallowed (eg. $name('test')).

What do you think about it? This can be implemented as a plugin that parses PHP file right after compilation.

There is nothing to implement to the parser :)

**Comment by webnull** [comment]
I have planned to replace current sandboxing with a plugin that will use PHP parser on RainTPL Next Generation. This could be easy to implement.

**Comment by rainphp** [comment]
Sure, whatever makes sense. I'd personally check on packager if exists
already a package that does that for you.

**Comment by webnull** [comment]
Yeah, I'm still talking about https://github.com/nikic/PHP-Parser

**Comment by webnull** [comment]
Partialy implemented in https://github.com/Panthera-Framework/raintpl3/commit/8bedcde688851d4658816ebe70c1e5a3a0975dc8

Requires a better whitelist + support for static includes (and detecting dynamic includes to ban them)
